tests/integration/extenders/MailTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Tests\integration\extenders;
+
+use Flarum\Extend;
+use Flarum\Api\Controller\ShowMailSettingsController;
+use Flarum\Mail\DriverInterface;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Flarum\Tests\integration\RetrievesAuthorizedUsers;
+use Flarum\Tests\integration\api\Controller\ApiControllerTestCase;
+use Flarum\User\User;
+use Illuminate\Contracts\Validation\Factory;
+use Illuminate\Support\MessageBag;
+use Swift_NullTransport;
+use Swift_Transport;
+
+class MailTest extends ApiControllerTestCase
```

Comment:
Please extend from `Flarum\Tests\integration\TestCase` and use the `request()` method.

I consider the `ApiControllerTestCase` deprecated (I guess I should mention that) and have been slowly migrating tests over to testing real requests - to further decouple the tests from the endpoint's implementation (i.e. a controller).